src/components/CrosshairToggle.jsx

```diff
@@ -1,29 +1,36 @@
 import React, { Component } from 'react';
 import {connect} from '../store/Connect';
-import {CrosshairIcon} from './Icons.jsx';
+import {CrosshairIcon, CrosshairOffIcon, CrosshairOnIcon, CrosshairTooltipIcon} from './Icons.jsx';
 import {Toggle} from './Form.jsx';
 
 const CrosshairToggle = ({
-    toggleCrosshair,
+    toggleState,
     crosshair,
+    state
 }) => (
     <div className="ciq-menu">
         <div className="cq-menu-btn">
             <Toggle
-                active={crosshair}
-                onChange={toggleCrosshair}
+                active={state != 0}
+                onChange={toggleState}
                 >
-                <CrosshairIcon
-                    className='ic-icon-with-sub'
-                    tooltip-title={t.translate('Crosshair')} />
+                {state == 0 ? <CrosshairOffIcon className='ic-icon-with-sub'
```

Comment:
I'm not a big fan of nested ternary operators; they are pretty hard to read. A possible solution is to use an array:

```js
const CrosshairIcon = [CrosshairOffIcon, CrosshairOnIcon, CrosshairTooltipIcon][state];
// ...then use as so:
<CrosshairIcon
    className='ic-icon-with-sub'
    tooltip-title={t.translate('Crosshair')}
/> 

```

Also with this way you reduce duplicate code.

Reply:
amazing solution, thank you